packages/grafana-runtime/src/services/templateSrv.ts

```diff
@@ -32,6 +32,16 @@ export interface TemplateSrv {
    * Update the current time range to be used when interpolating __from / __to variables.
    */
   updateTimeRange(timeRange: TimeRange): void;
+
+  /**
+   * Lists ad-hoc filters for a specific datasource
+   */
+  getAdhocFilters(datasourceName: string): AdHocVariableFilter[];
+
+  /**
+   * Extracts variable name from an expression like '${example}'
+   */
+  getVariableName(expression: string): string | undefined;
```

Comment:
I don't think we want to expose these, not like this at least

Reply:
I think we might need to expose something like this or an alternative approach to achieve this in a near future. Once we start to move core plugins out from the grafana/grafana repository we need to have a replacement.

Reply:
what is the use case for this function? 

We are doing lots of changes to templating at the moment

https://github.com/grafana/grafana/pull/65411
https://github.com/grafana/grafana/pull/65317 

as well as a new templating system in scenes :) 

So want limit any changes to the current system. 

Think  getAdhocFilters(datasourceName: string): should be fine, that is something some external plugins already use? 

 getVariableName(expression: string): string | undefined; is trickier, what about fieldPath, and formats? (in expressions like `${varName.fieldPath:format}`

Reply:
I didn't know about any future plans to change this part of the templating service so I opened the PR to get some feedback. I'm open to alternative ideas, and/or just shelving this, until the improvements to the system are finished.

(The use case for getVariableName is [here](https://github.com/grafana/iot-sitewise-datasource/pull/172). 
I added getAdHocFilters as well when I looked at internal plugins and saw that they were using it and, as @mckn mentioned, we will be moving out those and they will need to be exposed in some way eventually.)